agasc/supplement/utils.py

```diff
@@ -73,3 +79,60 @@ def get_supplement_table(name, agasc_dir=None, as_dict=False):
         out = Table(dat)
 
     return out
+
+
+def _load_or_create(filename, table_name):
+    try:
+        table = Table.read(filename, format='hdf5', path=table_name)
+    except OSError as e:
+        logger.debug(f'Creating agasc supplement table "{table_name}"')
+        if not filename.exists() or str(e) == f'Path {table_name} does not exist':
+            table = Table()
+        else:
+            raise
+    return table
+
+
+def save_version(filename, **kwargs):
+    """Save the version of a supplement table to the "versions" table.
+
+    Along with the version, the time of update is also added to another table called "last_updated"
+
+    Example usage::
+
+        from agasc.supplement.utils import save_version
+        save_version('agasc_supplement.h5', mags='4.10.2')
+
+    The different versions can be retrieved from the default supplement::
+
+        from agasc.supplement.utils import get_supplement_table
+        versions = get_supplement_table('versions')
+
+    :param filename: pathlib.Path
+    :param kwargs: dict
+        A dictionary of table name/version pairs,
+    """
+    import agasc
+    version_dtype = '<U32'
+    filename = Path(filename)
+
+    versions = _load_or_create(filename, 'versions')
+    last_updated = _load_or_create(filename, 'last_updated')
+
+    time = CxoTime().iso
+    kwargs.update({'supplement': agasc.__version__})
+    for key, value in kwargs.items():
+        pass
+        if key not in versions.colnames:
+            logger.debug(f'Adding "{key}" to agasc supplement "versions" table')
+            versions[key] = Column(data=['unknown'], dtype=version_dtype)
+        if key not in last_updated.colnames:
+            logger.debug(f'Adding "{key}" to agasc supplement "last_updated" table')
+            last_updated[key] = Column(data=['unknown'], dtype=version_dtype)
+        versions[key][:] = value
+        last_updated[key][:] = time
+
+    versions.write(str(filename), format='hdf5', path='versions',
```

Comment:
Astropy will accept a `Path`, no need for `str()`here.

Reply:
Fixed